### PR TITLE
Add MitoZ 3.0

### PR DESCRIPTION
**Pull request by linzhi2013**
Trying to ADD MitoZ (https://github.com/linzhi2013/MitoZ) to bioconda, thanks!

MitoZ: A toolkit for assembly, annotation, and visualization of animal mitochondrial genomes (https://doi.org/10.1093/nar/gkz173)

This pull request is an update to solve the problems mentioned in https://github.com/bioconda/bioconda-recipes/pull/34529.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by linzhi2013** [comment]
@BiocondaBot please add label

**Comment by dpryan79** [comment]
@BiocondaBot please fetch artifacts

File: recipes/mitoz/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - hmmer=3.1b2
+    - infernal=1.1.1
+    - openjdk
+    - perl=5.26
+    - perl-list-moreutils
+    - perl-params-validate
+    - perl-clone
+    - perl-bioperl
+    - python=3.6.7
+    - samtools=1.3.1
```
Comment (dpryan79) [inline recipes/mitoz/meta.yaml:44]: It needs these exact versions of python and samtools?

Comment (linzhi2013) [reply recipes/mitoz/meta.yaml:44]: Maybe Python 3.6 and other versions of samtools are okay, but I did not test them. Should we leave them to solve by conda? (although I do not expect too many differences for later versions of samtools). thanks!

**Comment by dpryan79** [comment]
@BiocondaBot please fetch artifacts

**Comment by dpryan79** [comment]
@BiocondaBot please fetch artifacts

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@BiocondaBot please merge

**Pull request merged**